File: sdk/runanywhere-swift/Sources/RunAnywhere/Public/Models/CostBreakdown.swift
import Foundation

/// Cost breakdown for generation
public struct CostBreakdown {
    /// Total cost in USD
    public let totalCost: Double

    /// Savings achieved by using on-device execution
    public let savingsAchieved: Double

    /// Cloud cost if it had been used
    public let cloudCost: Double?

    /// Device execution cost
    public let deviceCost: Double?

    public init(
        totalCost: Double,
        savingsAchieved: Double,
        cloudCost: Double? = nil,
        deviceCost: Double? = nil
    ) {
        self.totalCost = totalCost
        self.savingsAchieved = savingsAchieved
        self.cloudCost = cloudCost
        self.deviceCost = deviceCost
    }
}

File: sdk/runanywhere-android/src/main/java/com/runanywhere/sdk/models/CostBreakdown.kt
package com.runanywhere.sdk.models

/**
 * Cost breakdown for generation
 */
data class CostBreakdown(
    /**
     * Total cost in USD
     */
    val totalCost: Double,
    
    /**
     * Savings achieved by using on-device execution
     */
    val savingsAchieved: Double,
    
    /**
     * Cloud cost if it had been used
     */
    val cloudCost: Double? = null,
    
    /**
     * Device execution cost
     */
    val deviceCost: Double? = null
) 